protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/processinstance/ProcessEventRecord.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.zeebe.protocol.impl.record.value.processinstance;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.zeebe.msgpack.property.DocumentProperty;
+import io.zeebe.msgpack.property.LongProperty;
+import io.zeebe.msgpack.property.StringProperty;
+import io.zeebe.protocol.impl.encoding.MsgPackConverter;
+import io.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.zeebe.protocol.record.value.ProcessEventRecordValue;
+import io.zeebe.util.buffer.BufferUtil;
+import java.util.Map;
+import org.agrona.DirectBuffer;
+
+@SuppressWarnings("java:S2160")
+public final class ProcessEventRecord extends UnifiedRecordValue
+    implements ProcessEventRecordValue {
+  private final LongProperty scopeKeyProperty = new LongProperty("scopeKey", -1);
+  private final StringProperty targetElementIdProperty = new StringProperty("targetElementId", "");
```

Comment:
Do we need default values for `scopeKey` and `targetElementId`? 
I assume that these fields are always set. With default values, it is easier for the caller to forget to fill these fields.

Reply:
I've added processDefinitionKey and processInstanceKey, but for these kept a default value of -1 to indicate they are optional fields.